### ISSUE TITLE
nginx-sys does not build on rustc 1.98: suspicious_runtime_symbol_definitions on bindgen's libc declarations

### ISSUE THREAD

**Issue by u5surf** [nginx/ngx-rust#322]
### Describe the bug

`nginx-sys` no longer builds on the current stable toolchain. rustc 1.98.0 added `suspicious_runtime_symbol_definitions`, which fires on the libc declarations bindgen emits into `bindings.rs`, and CI runs with `RUSTFLAGS: '--deny warnings'`, so the warning is an error:

```
error: suspicious definition of the runtime `memcpy` symbol used by the standard library
     --> target/debug/build/nginx-sys-.../out/bindings.rs:16620:5
      |
16620 | /     pub fn memcpy(
16621 | |         __dest: *mut ::core::ffi::c_void,
16622 | |         __src: *const ::core::ffi::c_void,
16623 | |         __n: ::core::ffi::c_ulong,
16624 | |     ) -> *mut ::core::ffi::c_void;
      | |__________________________________^
      |
      = note: expected `unsafe extern "C" fn(*mut c_void, *const c_void, usize) -> *mut c_void`
              found    `unsafe extern "C" fn(*mut c_void, *const c_void, u64) -> *mut c_void`
      = note: `-D suspicious-runtime-symbol-definitions` implied by `-D warnings`
```

Six symbols are reported: `memcpy`, `memmove`, `memset`, `memcmp`, `bcmp` and `strlen`. All of them reach the bindings through `<string.h>`, which nginx's headers pull in; bindgen types their size arguments as `c_ulong`, and on 64-bit Linux the standard library expects `usize`. The two are the same width, which is why nothing is actually broken - the lint is about the declaration not matching the signature rustc knows for these symbols.

### To reproduce

Build with rustc 1.98.0 or newer and `RUSTFLAGS=--deny warnings`:

```sh
RUSTFLAGS='--deny warnings' cargo build -p nginx-sys --features vendored
```

Or just push anything: the `Test (Linux) (stable)`, `Test (MacOS)` and `Examples (Linux)` jobs all fail, while `Test (Linux) (1.85.0)` passes, since the MSRV pin predates the lint. A run showing exactly that is [here](https://github.com/u5surf/ngx-rust/actions/runs/33388916827).

### Expected behaviour

The crate builds on stable.

### Your environment

- ngx-rust `cda9d83` (current `main`) - not specific to any branch, `main` fails the same way
- rustc 1.98.0 (88d9e12ae 2026-08-18), as installed by `dtolnay/rust-toolchain@stable` on `ubuntu-latest`
- nginx 1.30.x via the `vendored` feature
- Reproduced on both the Linux and macOS jobs

Not reproducible on rustc 1.95.0, which is what I have locally - the lint does not exist there. The last green run on `main` is [31037465894](https://github.com/nginx/ngx-rust/actions/runs/31037465894) from 2026-08-05, before 1.98.0 was released.

### Additional context

These declarations are incidental. Nothing in ngx-rust calls them; they arrive because `build/wrapper.h` includes nginx's headers, which include `<string.h>`. Two ways out, both in `nginx-sys/build/main.rs` next to the existing `blocklist_item` calls:

```rust
// keep them out of the bindings entirely
.blocklist_function("memcpy")
.blocklist_function("memmove")
.blocklist_function("memset")
.blocklist_function("memcmp")
.blocklist_function("bcmp")
.blocklist_function("strlen")
```

or, if something is expected to use them:

```rust
.raw_line("#![allow(suspicious_runtime_symbol_definitions)]")
```

The blocklist reads better to me - these are libc functions that a caller should reach through `libc` or `core::ptr` rather than through nginx's bindings - but it is a wider change than it looks, since anything already calling `nginx_sys::memcpy` would stop compiling. Happy to send a PR for whichever you would prefer.
